app/services/remember_device_cookie.rb

```diff
@@ -0,0 +1,30 @@
+class RememberDeviceCookie
+  attr_reader :user_id, :created_at
+
+  def initialize(user_id:, created_at:)
+    @user_id = user_id
+    @created_at = created_at
+  end
+
+  def self.from_json(json)
+    parsed_json = JSON.parse(json)
+    new(
+      user_id: parsed_json['user_id'],
+      created_at: Time.zone.parse(parsed_json['created_at'])
+    )
+  end
+
+  def to_json
+    {
+      user_id: user_id,
+      created_at: created_at.iso8601,
```

Comment:
I'd add some more information here (based in part on https://www.immun.io/blog/be-careful-signing/encrypting-rails-cookies). At least a sentinel like `role: :remember_me,` (and check that the unencrypted value has this), and some random data to increase the entropy of the encrypted values (makes cracking the encryption key a bit more difficult since we otherwise have fairly easily guessable information).

Reply:
Good call, I'll pull that together

Reply:
LGTM